openquake/hazardlib/gsim/zhao_2016_volc_perg/volc_perg.py

```diff
@@ -0,0 +1,183 @@
+# under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# OpenQuake is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+from collections import OrderedDict
+
+from openquake.hazardlib.geo import polygon as pgn
+from openquake.hazardlib.geo.geodetic import npoints_between, distance
+
+
+def discretise_lines(ctx):
+    """
+    Discretize the line of minimum distance from each rupture surface to each
+    site and create mesh with matching discretization
+    :param ctx:
+        ctx: Context of ruptures and sites to compute ground-motions for
+    """
+    # Fix to zero to compute horz. dist. as in Zhao et al. 2016
+    fix_depth = 0
+
+    # Discretise line into 100 points
+    npoints_line = 100
+
+    # Get line from each rupture to each site
+    line_mesh = {}
+
+    for idx_path, val_site in enumerate(ctx.lon):
+
+        site_lon = ctx.lon[idx_path]
+        site_lat = ctx.lat[idx_path]
+        min_dist_rup_pnt_per_path_lon = ctx.clon[idx_path]
+        min_dist_rup_pnt_per_path_lat = ctx.clat[idx_path]
+
+        # Discretise shortest dist from rup srf to site and store as line
+        dsct_line = npoints_between(site_lon, site_lat, fix_depth,
+                                    min_dist_rup_pnt_per_path_lon,
+                                    min_dist_rup_pnt_per_path_lat,
+                                    fix_depth, npoints_line)
+
+        # Create mesh of discretized line
+        mesh_lons = dsct_line[0]
+        mesh_lats = dsct_line[1]
+        line_mesh[idx_path] = pgn.Mesh(mesh_lons, mesh_lats)
+        
+    return line_mesh
+
+
+def get_dist_traversed_per_zone(line_mesh, volc_pgn_store, polygon_per_zone,
+                                ctx):
+    """
+    Find the intercepts of the line from each rupture surface to each site within
+    each volcanic zone polygon (if present) and returns the distance traversed
+    per polygon
+    :param line_mesh:
+        line_mesh: Dict of meshes representing the line from each rupture to
+        each site
+    :param volc_pgn_store:
+        volc_pgn_store: Dict of zone ids + latitude and longitude of vertices
+        used to construct each polygon
+    :param polygon_per_zone:
+        polygon_per_zone: Polygon for each zone
+    :param ctx:
+        ctx: Context of ruptures and sites to compute ground-motions for
+    """
+    # Store the distanc per volc zone per travel path
+    dist_per_volc_zone_per_path = OrderedDict([(path_idx, {}) for path_idx,
```

Comment:
There is no need to use an OrderedDict since Python dictionaries are ordered since many years ago.

Reply:
Ordered dicts have been removed here now